examples/light-switch-app/efr32/BUILD.gn

```diff
@@ -92,14 +93,28 @@ if (silabs_board == "BRD4166A" || silabs_board == "BRD2601B" ||
 if (chip_enable_wifi) {
   wifi_sdk_dir = "${chip_root}/src/platform/EFR32/wifi"
   efr32_lwip_defs = [ "LWIP_NETIF_API=1" ]
-  efr32_lwip_defs += [
-    "LWIP_IPV4=1",
-    "LWIP_ARP=1",
-    "LWIP_ICMP=1",
-    "LWIP_DHCP=1",
-    "LWIP_IPV6_ND=1",
-    "LWIP_IGMP=1",
-  ]
+  if (lwip_ipv4) {
+    efr32_lwip_defs += [
+      "LWIP_IPV4=1",
+
+      # adds following options to provide
+      # them to .cpp source files
+      # flags ported from lwipopts file
+      # TODO: move lwipopts to one location
+      "LWIP_ARP=1",
+      "LWIP_ICMP=1",
+      "LWIP_IGMP=1",
+      "LWIP_DHCP=1",
+      "LWIP_DNS=0",
```

Comment:
If someone was previously using DNS for non-matter purposes, they don't have it anymore. The products should be able to override their onw lwipopts

Reply:
sure, I can omit `LWIP_DNS` from here, was moving the legacy options across as is.

Reply:
These options should all come from the lwipopts.h. That's the correct way. Forcing overrides on command line can have very unexpected results.

Reply:
@tcarmelveilleux This was already the way we were handling the setting of the defines prior to this PR. We can look into making this code improvement in another PR if that's OK since it affects more than just IPv4.

Reply:
@tcarmelveilleux I've logged an issue for making this code improvement: https://github.com/project-chip/connectedhomeip/issues/23346

Let me know if you can approve merging this PR for now.